src/txverificationqueue.h

```diff
@@ -0,0 +1,26 @@
+class CTxVerificationQueueEntry
+{
+private:
+    CTransaction tx;
+    CAmount nFee;
+    size_t nTxSize;
+    NodeId nodeId;
+public:
+    void setTX(const CTransaction& newTx);
+    void setNFee(CAmount newNFee);
+    void setNTxSize(size_t newNTxSize);
+    CTransaction& getTX();
+    void setNodeId(NodeId newNodeId);
+    NodeId getNodeId();
+};
+
+std::shared_ptr<CTxVerificationQueueEntry> createCTxVerificationQueueEntry(const CTransaction& tx, NodeId nodeId);
+
+class CTxVerificationQueue
+{
+public:
+    std::vector<std::shared_ptr<CTxVerificationQueueEntry>> vectorTX;;
```

Comment:
a deque could be use instead of a vector in order to achieve FIFO behaviour

Reply:
done
https://github.com/andreysobol/zen/commit/db174642ad56ef42a0c0b29aee97c4ee5e4151f9